src/windows/find_tools.rs

```diff
@@ -815,7 +815,20 @@ mod impl_ {
                 }
             }
             if version_file.is_empty() {
-                return None;
+                // If all else fails, manually search for bin directories.
+                let tools_dir: PathBuf = dir.join(r"VC\Tools\MSVC");
+                let version = tools_dir
+                    .read_dir()
+                    .ok()?
+                    .into_iter()
+                    .filter_map(Result::Ok)
+                    .filter_map(|file| {
+                        let name = file.file_name().into_string().ok()?;
```

Comment:
```suggestion
                        let name = OsString::into_string(file.file_name().).ok()?;
```

Reply:
The problem was the `file` is a `Result`. I also had to change `max_by_key` to `max_by` due to lifetime issues.

Reply:
Yeah once we bumped the msrv we can change it back